src/hotspot/cpu/aarch64/aarch64.ad

```diff
@@ -16551,6 +16569,27 @@ instruct array_equalsC(iRegP_R1 ary1, iRegP_R2 ary2, iRegI_R0 result,
   ins_pipe(pipe_class_memory);
 %}
 
+instruct arrays_hashcode(iRegP_R1 ary, iRegI_R2 cnt, iRegI_R0 result, immI basic_type,
+                         vRegD_V0 vtmp0, vRegD_V1 vtmp1, vRegD_V2 vtmp2, vRegD_V3 vtmp3,
+                         vRegD_V4 vtmp4, vRegD_V5 vtmp5, vRegD_V6 vtmp6, vRegD_V7 vtmp7,
+                         vRegD_V12 vtmp8, vRegD_V13 vtmp9, rFlagsReg cr)
```

Comment:
It would be much better to pass all the vtemp registers to the `arrays_hashcode()`, then assert the registers are correct.

See `LOOKUP_SECONDARY_SUPERS_TABLE_REGISTERS` in `MacroAssembler`. It is asserted in the stub and in the inlined code. If this had been done in the patch that had to be backed out, it wouldn't have had to be backed out.

Reply:
Agreed. Would be nice to have an established mechanism for this though. I think we could do better than a macro but it should do as well.

Reply:
> Agreed. Would be nice to have an established mechanism for this though. I think we could do better than a macro but it should do as well.

I'm not sure we could do _much_ better. A great advantage of using a macro is that it enforces consistent naming across the stub and the inlined code.

Reply:
@theRealAph , could you check this?

Reply:
Hi @theRealAph , could you re-review this?